cmd/posix.go

```diff
@@ -384,11 +384,60 @@ func (s *posix) CrawlAndGetDataUsage(ctx context.Context, cache dataUsageCache)
 			return 0, nil
 		}
 
+		if cache.Info.lifeCycle != nil {
+			objName, err := filepath.Rel(filepath.Join(s.diskPath, cache.Info.Name), item.Path)
+			if err != nil {
+				logger.LogIf(ctx, err)
+				return meta.Stat.Size, nil
+			}
+			objName = filepath.Dir(objName)
+
+			action := cache.Info.lifeCycle.ComputeAction(objName, meta.Meta[xhttp.AmzObjectTagging], meta.Stat.ModTime)
+			switch action {
+			case lifecycle.DeleteAction:
+			default:
+				// No action.
+				return meta.Stat.Size, nil
+			}
+
+			// These (expensive) operations should only run on items we are likely to delete.
+			// Load to ensure that we have the correct version and not an unsynced version.
+			obj, err := globalObjectAPI.GetObjectInfo(ctx, cache.Info.Name, objName, ObjectOptions{})
```

Comment:
It cannot be accessed in this manner you would have to use `newObjectLayerFn()`  and obtain the value instead to be safe.